sympy/polys/rings.py

```diff
@@ -1268,7 +1281,6 @@ def square(self):
 
     def __divmod__(p1, p2):
         ring = p1.ring
-        p = ring.zero
 
         if not p2:
```

Comment:
maybe this should be `if p2 == zero` and `p = ring.zero` above should be `zero = ring.zero`? ping @jksuom

Reply:
below, too

Reply:
This code looks rather like a first draft to me. There is an invocation of `__rdivmod__`, that is not implemented. As to `if not p2`, I suppose that it was intended as a quick first step that would work with all PolyElements regardless of their rings. (A `PolyElement` is a `dict`, and an empty `dict` can be quickly detected.) So it is probably preferable to `if p2 == ring.zero` that would only work with polynomials over the base ring of `p1`.